src/App.tsx

```diff
@@ -14,11 +15,87 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+const App: React.FC = () => {
+  const [goods, setGoods] = useState(goodsFromServer);
+  const [isShowList, setShowList] = useState(true);
+
+  const startShowList = () => {
+    setShowList((current) => !current);
+  };
```

Comment:
you can read about [useCallback](https://reactjs.org/docs/hooks-reference.html#usecallback)